infosys-display/src/db.rs

```diff
@@ -0,0 +1,82 @@
+extern crate config;
+extern crate postgres;
+
+use self::postgres::Connection;
+use self::postgres::TlsMode;
+use self::postgres::tls::native_tls::NativeTls;
+
+static SQL_CREATE_SCHEDULE: &'static str = "CREATE TABLE schedule (
+    timeslot        TIME NOT NULL PRIMARY KEY,
+    message_id      INTEGER NOT NULL,
+    FOREIGN KEY     (message_id) REFERENCES messages(id)
+);";
+
+static SQL_CREATE_MESSAGE: &'static str = "CREATE TABLE messages (
+    id SERIAL PRIMARY KEY
+);";
+
+static SQL_CREATE_STRING: &'static str = "CREATE TABLE strings (
+    id              INTEGER NOT NULL,
+    message_id      INTEGER NOT NULL,
+    mode            VARCHAR(32) NOT NULL,
+    data            TEXT NOT NULL,
+    PRIMARY KEY     (id, message_id),
+    FOREIGN KEY     (message_id) REFERENCES messages(id)
+);";
+
+static SQL_DEFAULT_VALUES: &'static str = "
+INSERT INTO messages (id) VALUES (0);
+
+INSERT INTO schedule (timeslot, message_id) VALUES ('00:00:00', 0);
+
+INSERT INTO strings (id, message_id, mode, data) VALUES (1, 0, 'STANDARD_HOLD', 'Welcome to CSH!');";
+
+pub fn db_init(settings: &config::Config) ->Connection{
+    let negotiator = NativeTls::new().unwrap();
+    let con = Connection::connect(
+                settings.get_str("dbstring").unwrap(),
+                TlsMode::Require(&negotiator)
+    ).unwrap();
+
+    let mut init = true;
+    init = init && check_or_create_db(&con, "messages", SQL_CREATE_MESSAGE);
```

Comment:
nit: This feels like it could be more cleanly chained together by using an `Option` or `Result` type